Lib/importlib/machinery.py

```diff
@@ -1,19 +1,19 @@
 """The machinery of importlib: finders, loaders, hooks, etc."""
 
-from ._bootstrap import ModuleSpec
-from ._bootstrap import BuiltinImporter
-from ._bootstrap import FrozenImporter
-from ._bootstrap_external import (SOURCE_SUFFIXES, DEBUG_BYTECODE_SUFFIXES,
-                     OPTIMIZED_BYTECODE_SUFFIXES, BYTECODE_SUFFIXES,
-                     EXTENSION_SUFFIXES)
-from ._bootstrap_external import WindowsRegistryFinder
-from ._bootstrap_external import PathFinder
-from ._bootstrap_external import FileFinder
-from ._bootstrap_external import SourceFileLoader
-from ._bootstrap_external import SourcelessFileLoader
-from ._bootstrap_external import ExtensionFileLoader
-from ._bootstrap_external import AppleFrameworkLoader
-from ._bootstrap_external import NamespaceLoader
+from ._bootstrap import ModuleSpec  # noqa
+from ._bootstrap import BuiltinImporter  # noqa
+from ._bootstrap import FrozenImporter  # noqa
+from ._bootstrap_external import (SOURCE_SUFFIXES, DEBUG_BYTECODE_SUFFIXES,  # noqa
+                     OPTIMIZED_BYTECODE_SUFFIXES, BYTECODE_SUFFIXES,  # noqa
+                     EXTENSION_SUFFIXES)  # noqa
+from ._bootstrap_external import WindowsRegistryFinder  # noqa
+from ._bootstrap_external import PathFinder  # noqa
+from ._bootstrap_external import FileFinder  # noqa
+from ._bootstrap_external import SourceFileLoader  # noqa
+from ._bootstrap_external import SourcelessFileLoader  # noqa
+from ._bootstrap_external import ExtensionFileLoader  # noqa
+from ._bootstrap_external import AppleFrameworkLoader  # noqa
+from ._bootstrap_external import NamespaceLoader  # noqa
```

Comment:
I don't mind adding one or two `noqa`s to a file, but for files like this, adding so many feels very ugly to me. I'd prefer it if we instead considered adding `__all__` -- a linter will not complain about any "unused" imports that are included in `__all__`. If we're okay with comments that are specific to a certain tool, `# ruff: noqa: F401` on a line by itself will also mean that Ruff will ignore all F401 violations for that file.

Reply:
Beware, adding to `__all__` is a semantic change.  But in general when wanting to apply linter comments to a long sequence of lines or any statement spanning multiple lines, using a block comment enable/disable check semantic is more maintainable.

```
# pylint: disable=thing
lots of stuff that
violates thing.
# pylint: enable=thing
```

(in whatever tooling common way ruff supports for block disable/enable)

Reply:
> Beware, adding to `__all__` is a semantic change.

Agreed that it's strictly-speaking backwards incompatible (since `*` imports won't work the same), but [it is a change we've made to long-established modules in the past](https://github.com/python/cpython/pull/30003).

Reply:
I don't believe Ruff supports switching off a lint rule for a specific block of code (just for a specific line or a specific file), but I might be wrong.

Reply:
I reverted changes in 8 files to keep the less controversial `# noqa` changes :-)